File: exercises/exercise-1-testing-java/src/main/java/de/emaarco/example/adapter/inbound/zeebe/bike/SendBikeConfirmationMailWorker.java
package de.emaarco.example.adapter.inbound.zeebe.bike;

import de.emaarco.example.adapter.process.generated.BikeSubscriptionSignupProcessApi;
import de.emaarco.example.application.port.inbound.bike.SendBikeConfirmationMailUseCase;
import de.emaarco.example.domain.bike.BikeSubscriptionId;
import io.camunda.client.annotation.JobWorker;
import io.camunda.client.annotation.Variable;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;
import org.springframework.stereotype.Component;

import java.util.UUID;

@Component
public class SendBikeConfirmationMailWorker {

    private static final Logger log = LoggerFactory.getLogger(SendBikeConfirmationMailWorker.class);

    private final SendBikeConfirmationMailUseCase useCase;

    public SendBikeConfirmationMailWorker(SendBikeConfirmationMailUseCase useCase) {
        this.useCase = useCase;
    }

    @JobWorker(type = BikeSubscriptionSignupProcessApi.TaskTypes.BIKE_SEND_CONFIRMATION_MAIL)
    public void handle(@Variable UUID subscriptionId) {
        log.info("Sending bike confirmation mail for subscription: {}", subscriptionId);
        useCase.sendConfirmationMail(new BikeSubscriptionId(subscriptionId));
    }

}

File: exercises/exercise-1-testing-kotlin/src/main/kotlin/de/emaarco/example/adapter/inbound/zeebe/bike/SendBikeConfirmationMailWorker.kt
package de.emaarco.example.adapter.inbound.zeebe.bike

import de.emaarco.example.adapter.process.generated.BikeSubscriptionSignupProcessApi
import de.emaarco.example.application.port.inbound.bike.SendBikeConfirmationMailUseCase
import de.emaarco.example.domain.bike.BikeSubscriptionId
import io.camunda.client.annotation.JobWorker
import io.camunda.client.annotation.Variable
import mu.KotlinLogging
import org.springframework.stereotype.Component
import java.util.UUID

@Component
class SendBikeConfirmationMailWorker(
    private val useCase: SendBikeConfirmationMailUseCase
) {
    private val log = KotlinLogging.logger {}

    @JobWorker(type = BikeSubscriptionSignupProcessApi.TaskTypes.ACTIVITY_SEND_CONFIRMATION_MAIL)
    fun handle(@Variable subscriptionId: UUID) {
        log.info { "Sending bike confirmation mail for subscription: $subscriptionId" }
        useCase.sendConfirmationMail(BikeSubscriptionId(subscriptionId))
    }
}
